day1/part2.py

```diff
@@ -1,43 +1,31 @@
-with open('input.txt', 'r') as f:
-    numbers = ('one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine')
-    lines = list(map(str.rstrip, f.readlines()))
-    num_of_line = 0
-    counter = 0
-    for line in lines:
-        counter += 1
-        for character in line:
-            if character.isdigit():
-                first = {character: line.index(character)}
-                break
-        for number in numbers:
-            if line.find(number) != -1:
-                first.update({number: line.index(number)})
-
-        for character in line[::-1]:
+def find_digit(ln: str, directio: int, numbers, my_dist):
+    if numbers:
+        for character in ln[::directio]:
             if character.isdigit():
-                last = {character: line[::-1].index(character)}
-                break
-        for number in numbers:
-            if line.find(number) != -1:
-                last.update({number: line[::-1].index(number[::-1])})
+                my_dist.update({character: ln[::directio].index(character)})
+        for num in number:
+            if ln.find(num) != -1:
+                my_dist.update({num: ln[::directio].index(num[::directio])})
 
 
-        tmp = min(first, key=first.get)
-        if tmp in numbers:
-            left_number = numbers.index(tmp) + 1
-        else:
-            left_number = tmp
-        tmp = min(last, key=last.get)
-        if tmp in numbers:
-            right_number = numbers.index(tmp) + 1
-        else:
-            right_number = tmp
-
-        print(str(left_number) + str(right_number))
-        num_of_line += int(str(left_number) + str(right_number))
-        first.clear()
-        last.clear()
-    print(num_of_line)
-    print(counter)
+def get_num(my_dist):
+    tmp = min(my_dist, key=my_dist.get)
+    if tmp in number:
+        return number.index(tmp) + 1
+    else:
+        return tmp
 
 
+number = ('one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine')
+with open('input.txt', 'r') as f:
+    lines = list(map(str.rstrip, f.readlines()))
+    total = 0
+    for line in lines:
+        first: dict = {}
+        last: dict = {}
+        find_digit(line, 1, True, first)
```

Comment:
This part can be solved better, Try to think about the problem and simplify your solution. Also the variables are all over the places and do not seem to be needed. Do not create a variable just to return get rid of it after one use just pass in the result of the function call. IMHO.